driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1162,6 +1162,8 @@ private Manager(String clusterName, List<InetSocketAddress> contactPoints, Confi
 
             this.metrics = configuration.getMetricsOptions() == null ? null : new Metrics(this);
             this.listeners = new CopyOnWriteArraySet<Host.StateListener>(listeners);
+
+            this.scheduledTasksExecutor.scheduleWithFixedDelay(new CleanupIdleConnectionsTask(), 10, 10, TimeUnit.SECONDS);
```

Comment:
Looking at scheduledTasksExecutor, it's primary use before was for refreshNodeInfo and the executor is single threaded to serialize node refresh requests.  With adding CleanupIdleConnectionsTask to the same executor, there is a small concern that the control connection can be unstable and the queries in refreshNodeInfo are delayed, delaying  idle connections from being cleaned up.  Not really that big of a concern but for that reason might be nice to create a separate ScheduledTaskExecutor for this.